diff --git a/js/private/js_helpers.bzl b/js/private/js_helpers.bzl
index f0dcdef..2652273 100644
--- a/js/private/js_helpers.bzl
+++ b/js/private/js_helpers.bzl
@@ -272,6 +272,8 @@ def gather_files_from_js_infos(
     files_depsets = []
 
     for target in targets:
+        info = target[DefaultInfo]
+        files_depsets.append(info.data_runfiles.files)
         if JsInfo in target:
             js_info = target[JsInfo]
 
